### PR TITLE
Consume Makefile in CircleCI config

### PR DESCRIPTION
**Pull request by wpcarro**
- Couple the CircleCI configuration to some of the Makefile commands
- Support new `opam-deps` command in the Makefile
- Change `make docs` -> `make documentation`

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -41,9 +41,10 @@ jobs:
             - v2-native-dependencies-
       # m4 is a system dependency required by conf-m4 -> ocamlfind -> fmt -> alcotest
       - run: sudo apt-get install -y m4
-      - run: opam update && opam install alcotest base dune junit junit_alcotest -y
-      - run: opam config exec -- dune build
-      - run: opam config exec -- dune runtest -f
+      - run: make opam-deps
```
Comment (pbiggar) [inline .circleci/config.yml]: Perhaps call this deps-native

File: .circleci/config.yml [hunk 2]
```diff
@@ -18,8 +18,8 @@ jobs:
             - v1-dependencies-{{ .Branch }}-
             - v1-dependencies-
       - run: npm install
```
Comment (pbiggar) [inline .circleci/config.yml]: Call `npm install` `deps-bs`?

**Comment by wpcarro** [comment]
Updated `deps-native` and `deps-bs`. Will need to take a closer look at the broken build and GH Pages integration tomorrow. Calling it for the night. Talk soon!

**Comment by wpcarro** [comment]
@pbiggar fixed the build. I'm dropping support for building and publishing the documentation from this PR. I think it'll take a bit more effort than I anticipated.

I'd like to get this in and perhaps if I have more time, I can give the documentation another look later on in a different PR.

**Comment by pbiggar** [comment]
Good idea.

**Pull request merged**